flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DefaultDispatcherResourceManagerComponentFactory.java

```diff
@@ -182,26 +182,25 @@ public DispatcherResourceManagerComponent create(
 
 			final HistoryServerArchivist historyServerArchivist = HistoryServerArchivist.createHistoryServerArchivist(configuration, webMonitorEndpoint);
 
-			jobManagerMetricGroup = MetricUtils.instantiateJobManagerMetricGroup(
```

Comment:
We remains a `jobManagerMetricGroup` variable above. IIUC we change `jobManagerMetricGroup` to `JobManagerMetricGroupFactory` so that we can remove that variable.

Reply:
True, I will remove it.